### PR TITLE
Fixes Multi Hit moves removing destiny bond flag in the middle of attack

### LINKED ISSUE #5204: Destiny Bond is bugged with multi-hit moves
### Description

Currently, Destiny Bond will fail to take down attackers if the defending Destiny Bond user is knocked out by a multi-hit move after the first hit.

This is what the bug looks like in action:
https://github.com/user-attachments/assets/802b396f-5439-4868-85a6-8bb020bdcd55

As mentioned previously, Destiny Bond works correctly if the first hit is the hit that knocks out the Destiny Bond user:
https://github.com/user-attachments/assets/06936e0d-737d-4bbe-9018-abfdd0bd7574

Tested on clean latest expansion.

### Version

1.9.0 (Latest release)

### Upcoming/master Version

_No response_

### Discord contact info

sarnnn

### PR DESCRIPTION
**Pull request by AlexOn1ine**
Fixes #5204

### CODE REVIEW

**Review by Pawkkie — APPROVED** [review]
Loving these micro fixes

File: src/battle_script_commands.c [hunk 1]
```diff
@@ -4686,7 +4686,8 @@ static void MoveValuesCleanUp(void)
     gIsCriticalHit = FALSE;
     gBattleScripting.moveEffect = 0;
     gBattleCommunication[MISS_TYPE] = 0;
-    gHitMarker &= ~HITMARKER_DESTINYBOND;
+    if (!gMultiHitCounter)
```
Comment (AsparagusEduardo) [inline src/battle_script_commands.c:4689]: Potential improvement for the future: `if (gMultiHitCounter > 1)`, as setting multi-hit to 1 could potentially make Destiny Bond not work here.

Comment (AlexOn1ine) [reply src/battle_script_commands.c:4689]: > unless it's reduced before it reaches this?
Yes, it has to be reduced to zero. Otherwise it doesn't work

Comment (AlexOn1ine) [reply src/battle_script_commands.c:4689]: but `gMultiHitCounter > 0` would still look better imo

Comment (AsparagusEduardo) [reply src/battle_script_commands.c:4689]: Understood :)

Comment (AlexOn1ine) [reply src/battle_script_commands.c:4689]: let's keep `if (!gMultiHitCounter)` for consistency. It is used in a lot of places like that.

**Pull request merged**